tests/routes/test_auth.py

```diff
@@ -2,113 +2,165 @@
 
-    @patch.dict(os.environ, {"AUTH0_DOMAIN": ""})
-    @patch.dict(os.environ, {"AUTH0_CLIENT_ID": ""})
-    def test_logout(self):
-        response = self.client.get("/auth/logout")
+    def tearDown(self):
+        self.ctx.pop()
+
+    @patch("join_github_app.main.routes.auth.oauth.auth0.authorize_redirect")
+    def test_login_route(self, mock_authorize_redirect):
+        mock_response = Response(status=302, headers={'Location': 'mock://auth0.redirect'})
+        mock_authorize_redirect.return_value = mock_response
+
+        response = self.client.get("/auth/login")
         self.assertEqual(response.status_code, 302)
-        self.assertIn("Location", response.headers)
-        self.assertIn("v2/logout", response.headers["Location"])
+        mock_authorize_redirect.assert_called_once()
 
+    @patch("os.getenv")
+    def test_logout_route(self, mock_getenv):
+        mock_getenv.side_effect = lambda var: {"AUTH0_DOMAIN": "test.auth0.com", "AUTH0_CLIENT_ID": "test_client_id"}.get(var)
+
+        with self.client as c:
+            with c.session_transaction() as sess:
+                sess['user'] = {'some': 'data'}
+
+            response = c.get("/auth/logout")
+
+            self.assertEqual(response.status_code, 302)
+
+            with c.session_transaction() as sess:
+                self.assertNotIn('user', sess)
+
+        self.assertIn('auth0.com/v2/logout', response.headers['Location'])
+
+    @patch("join_github_app.main.routes.auth.render_success_page")
+    @patch("join_github_app.main.routes.auth.send_github_invitation", return_value=True)
+    @patch("join_github_app.main.routes.auth.process_user_session", return_value=True)
+    @patch("join_github_app.main.routes.auth.get_token")
+    def test_callback_route_success(self, mock_get_token, mock_process_user_session,
+                                    mock_send_github_invitation, mock_render_success_page):
+        mock_get_token.return_value = {'userinfo': {'email': 'test@example.com'}}
+        mock_render_success_page.return_value = Response(status=200)
+
+        response = self.client.get("/auth/callback")
+        self.assertEqual(response.status_code, 200)
+        mock_get_token.assert_called_once()
+        mock_process_user_session.assert_called_once()
+        mock_send_github_invitation.assert_called_once()
+        mock_render_success_page.assert_called_once()
+
+    @patch("join_github_app.main.routes.auth.render_error_page", return_value=Response(status=500))
+    @patch("join_github_app.main.routes.auth.get_token")
+    def test_callback_route_token_failure(self, mock_get_token, mock_render_error_page):
+        mock_get_token.side_effect = AuthTokenError("Token error")
+
+        response = self.client.get("/auth/callback")
+        self.assertEqual(response.status_code, 500)
+        mock_get_token.assert_called_once()
+        mock_render_error_page.assert_called_once()
+
+    @patch("join_github_app.main.routes.auth.ALLOWED_EMAIL_DOMAINS",
```

Comment:
```suggestion
    @patch("join_github_app.main.routes.auth.ALLOWED_EMAIL_DOMAINS",
```
Presumably this doesn't need a `clear=True` arg as `@patch.dict` does?